ironmon_tracker/data/RandomizerLog.lua

```diff
@@ -0,0 +1,2049 @@
+
+function RandomizerLog.parseTrainers(logLines)
+	local sectorHeader = RandomizerLog.Patterns.getSectorHeaderPattern(RandomizerLog.Sectors.Trainers.Header)
+	if not RandomizerLog.seekSectorStart(logLines, sectorHeader) then
+		return
+	end
+
+	-- Parse the sector
+	local line = table.remove(logLines, 1)
+	while line ~= nil do
+		local trainer_num, trainername, party = string.match(line, RandomizerLog.Sectors.Trainers.NextTrainerPattern)
+		trainer_num = tonumber(RandomizerLog.formatInput(trainer_num) or "") -- nil if not a number
+		trainername = RandomizerLog.formatInput(trainername)
+
+		-- If nothing matches, end of sector
+		if trainer_num == nil or trainername == nil or party == nil then
+			table.insert(logLines, 1, line)
+			return
+		end
+
+		RandomizerLog.Data.Trainers[trainer_num] = {
+			name = trainername, -- likely in the form of TRAINER CLASS + TRAINER NAME
+			party = {},
+		}
+
+		for partypokemon in string.gmatch(party, RandomizerLog.Sectors.Trainers.PartyPattern) do
+			local pokemon, helditem, level = string.match(partypokemon, RandomizerLog.Sectors.Trainers.PartyPokemonPattern)
+			pokemon = RandomizerLog.formatInput(pokemon)
+			helditem = RandomizerLog.formatInput(helditem)
+			level = tonumber(RandomizerLog.formatInput(level) or "") or 0 -- nil if not a number
+			if helditem == "" then
+				helditem = nil -- don't waste storage if empty
+			end
+
+			if pokemon ~= nil and RandomizerLog.PokemonNameToIdMap[pokemon] ~= nil then
+				local partyPokemon = {
+					pokemonID = RandomizerLog.PokemonNameToIdMap[pokemon],
+					helditem = helditem,
+					level = level,
+				}
+				table.insert(RandomizerLog.Data.Trainers[trainer_num].party, partyPokemon)
+			end
+		end
+		line = table.remove(logLines, 1)
+	end
+end
+-- Utils.printDebug("#%s: %s >%s< %s", trainer_num, pokemon or "N/A", helditem or "N/A", level or 0)
+
+-- Currently unused
+function RandomizerLog.parsePickupItems(logLines)
+
+end
+
+-- DEBUG HELPER
+function RandomizerLog.printPokemonIds()
+	local count = 0
+	local mons = {}
+	for id, pokemon in ipairs(PokemonData.Pokemon) do
+		if PokemonData.isValid(id) then
+			table.insert(mons, string.format('["%s"] = %s,', pokemon.name:lower(), id))
+			count = count + 1
+		end
+		-- Print 20 lines at a time
+		if count >= 20 then
+			print(table.concat(mons, "\n"))
+			mons = {}
+			count = count - 20
+		end
+	end
+	if count >= 1 then
+		print(table.concat(mons, "\n"))
+		mons = {}
+	end
+end
+
+-- DEBUG HELPER
+function RandomizerLog.printMoveIds()
+	local count = 0
+	local moves = {}
+	for id, move in ipairs(MoveData.Moves) do
+		if MoveData.isValid(id) then
+			table.insert(moves, string.format('["%s"] = %s,', move.name:lower(), id))
+			count = count + 1
+		end
+		-- Print 20 lines at a time
+		if count >= 20 then
+			print(table.concat(moves, "\n"))
+			moves = {}
+			count = count - 20
+		end
+	end
+	if count >= 1 then
+		print(table.concat(moves, "\n"))
+		moves = {}
+	end
+end
+
+function RandomizerLog.setupMappings()
```

Comment:
potential big thing to note with this, these mappings are for the english names so likely won't work on non-english roms (which aren't officially supported on the randomiser but do still often work)

could be fine for now, as long as it's noted, but yeah may need to come up with a more compact/elegant solution that allows substituting the names via the language data we already have. I guess easiest solution could be straight-up replacing these mappings when setting up language data for the language loaded, depending on the load order of scripts?

Reply:
Agree. I have some side notes that logviewer really only works for the currently loaded game (cant load a Ruby log in a FireRed game). And also aware that many of the randomizer logs are likely in english only. I'll make a note of this in the release.